enhancements/upgrade-readiness.md

```diff
@@ -0,0 +1,580 @@
+  }
+}
+```
+
+When this annotation is present on a CRD, OLM will append the following `Probe` CR to the `installPlan`:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: Probe
+metadata:
+  name: foo-operator.v1.0.0
+  namespace: operators
+spec:
+  manager: crdAnnotations # Changes to the CRD's annotations should override the spec. OLM will watch the CRD for changes to the "OLM Supported Condition" annotations and update the spec accordingly.
+  probeResources:
+  - resource: foo.example.com
+    upgradeable: "!Migrating"
+    important: "BadConnectivity || UnhealthyDatabase"
+```
+
+When the`probe` resource above is created, OLM will spin up an SPO pod which collects the information from conditions of `foo.example.com` CRs and communicate it to OLM via the `probe`'s status accordingly. If an operator were to define multiple resources that it communicates its status through, a single SPO pod would monitor all of those resources. The permissions of the probe will match the same permissions of the operator.
+
+>Note: If the probe's spec is nil, SPO will not create a pod and the operator will be expected to update the CRs status itself.
+
+In this example, the SPO will be responsible for maintaining the following fields in the status of the `probe` resource:
+
+```yaml
+status:
+  conditions:
+    upgradeable: # A boolean that represents if the operator may be upgraded - aggregated from all probeResources below.
+  probeResources: # The list of probe resources.
+```
+
+Consider an upgrade where the a single `Foo` CR exists on cluster:
+
+```yaml
+apiVersion: foo.example.com/v1
+kind: Foo
+metadata:
+  name: foo-example
+  namespace: default
+spec:
+  ...
+  ...
+  ...
+status:
+  conditions:
+  - type: Migrating # Condition 1
+    status: true
+    reason: "CriticalProcess"
+    message: "Migrating old database schema to new format"
+  - type: UnhealthyDatabase # Condition 2
+    status: true
+    reason: "LivenessProbeFailure"
+    message: "The foo-database failed its liveness probe."
+  - type: BadConnectivity # Condition 3
+    status: true
+    reason: "404"
+    message: "Unable to reach the database"
+  - type: SomethingUnrelated # Condition 4
+    status: true
+    reason: "Unmatched"
+    message: "OLM doesn't care about this condition"
+```
+
+When SPO registers this CR, the Probe CR will be updated like so:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: Probe
+metadata:
+  name: foo-operator.v1.0.0 # Same name as the CSV.
+  namespace: operators
+spec:
+  manager: crdAnnotations # When set to `crdAnnotations` OLM will override any changes to the probe's spec to match those defined via annotations on the CRDs described above.
+  probeResources:
+  - name: foo.example.com
+    upgradeable: "!Migrating"
+    important: "BadConnectivity || UnhealthyDatabase"
+status:
+  conditions:
+  - type: Upgradeable # Overall Upgradeable condition
+    status: false
```

Comment:
nit: `status: false`.  This is a string-ish type, not a YAML boolean.

Reply:
Thanks - I have made local changes which corrects this in all places.